indico/modules/events/contributions/models/contributions.py

```diff
@@ -510,7 +511,19 @@ def enabled_editables(self):
             return []
 
         enabled_editable_types = editing_settings.get(self.event, 'editable_types')
-        return [editable for editable in self.editables if editable.type.name in enabled_editable_types]
+        enabled_editables = [editable for editable in self.editables if editable.type.name in enabled_editable_types]
+        order = list(EditableType)
+        return sorted(enabled_editables, key=lambda editable: order.index(editable.type))
+
+    @property
+    def has_accepted_editables(self):
+        for editable in self.enabled_editables:
+            if editable.revisions:
+                latest_revision = editable.revisions[-1]
+                latest_state = latest_revision.final_state or latest_revision.initial_state
+                if latest_state.name == 'accepted':
+                    return True
+        return False
```

Comment:
return any(e.published_revision_id is not None for e in self.enabled_editables)

The state doesn't matter (even though it should always be accepted) - what matters is that the revision is marked as published. With the microservice it is actually possible that a revision is accepted but not published yet ("awaiting QA")